### DISCUSSION TITLE
Adding JSON-LD to element pages

### DISCUSSION THREAD

**Discussion by Silvanoc** [linkml/linkml#1560] [Ideas]
As of now the webpages being generated for the different LinkML elements (for example, classes, slots,...) are only usaful for human beings, since they are easy HTML pages. Wouldn't it be useful to have the same information in **JSON-LD** format for easier machine processing?

Although all the information that would appear is available in the schema in LinkML format (or its JSON-LD equivalent), each element webpage could give a subset of the schema in JSON-LD format with the element properties and linking to the other webpages.

See following Python example to get the JSON-LD corresponding to the schema.org IndividualProduct entry:
```python
import requests
from bs4 import BeautifulSoup as bs
from urllib.parse import urljoin

# URL of the web page you want to extract
url = "https://schema.org/IndividualProduct"

# initialize a session
session = requests.Session()
# set the User-agent as a regular browser
session.headers["User-Agent"] = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.157 Safari/537.36"

# get the HTML content
html = session.get(url).content

# parse HTML using beautiful soup
soup = bs(html, "html.parser")

for script in soup.find_all("script"):
    if script.attrs.get("type") == "application/ld+json":
        print(script.string)
```

**Comment by cmungall** [comment]
I'm surprised schema.org doesn't support conneg?

```
curl https://schema.org/IndividualProduct  -H 'accept: application/ld+json'
Warning: Binary output can mess up your terminal. Use "--output -" to tell
Warning: curl to output it to your terminal anyway, or consider "--output
Warning: <FILE>" to save to a file.
```

We could in theory support conneg for LinkML schemas but the way most schemas are deployed (static sites / github pages) means this would mean pre-generating json-ld for every element and adding it to the repo.

I'm not completely convinced there is a use case for machine processing individual elements by using BS to scrape content from HTML.. it seems a pretty tortuous and non-robust path to get something we make create pains to make machine accessible via more conventional means.

But it wouldn't be hard to embed various translations of schema elements inside `<script>` tags. We may want to make this opt-in at first as there may be consequences for things like search (already some LinkML schemas are hard to search using out-the-box github pages indexing strategies)

I like how schema.org embeds examples in different formats:

<img width="1022" alt="image" src="https://github.com/linkml/linkml/assets/50745/7e4acb41-17d3-4812-9304-4e6df8b10234">

This is something we should be able to do now we have a standardized way of including examples with a schema

**Reply by Silvanoc** [reply]
You're probably right and there's no real use-case for it.

Though I would expect the whole schema to get embedded into one of the pages in YAML and/or JSON-LD format, but it doesn't appear to happen. As of now I'm copying both schema formats into the pages directory to make it downloadable.
